app/models/package_manager/maven.rb

```diff
@@ -109,45 +110,58 @@ def self.mapping(raw_project, depth = 0)
       nil
     end
 
-    def self.mapping_from_pom_xml(version_xml, depth = 0)
-      xml = if version_xml.respond_to?("project")
-              version_xml.project
-            else
-              version_xml
-            end
-
-      parent = {
-        description: nil,
-        homepage: nil,
-        repository_url: "",
-        licenses: "",
-        properties: {},
-      }
-      if xml.locate("parent").present? && depth < MAX_DEPTH
-        group_id = extract_pom_value(xml, "parent/groupId")&.strip
-        artifact_id = extract_pom_value(xml, "parent/artifactId")&.strip
-        version = extract_pom_value(xml, "parent/version")&.strip
-        if group_id && artifact_id && version
-          parent = mapping_from_pom_xml(
-            get_pom(group_id, artifact_id, version),
-            depth + 1
-          )
+    def self.get_pom_documents(group_id, artifact_id, version, depth = 0)
+      return [] if depth >= MAX_DEPTH
+
+      pom_document = get_pom(group_id, artifact_id, version)
+      if pom_document.respond_to?("project")
+        pom_document = pom_document.project
+      end
+
+      result = [pom_document]
+
+      # Check if the POM document has a parent
+      if pom_document.locate("parent").present?
+        parent_group_id = extract_pom_value(pom_document, "parent/groupId")&.strip
+        parent_artifact_id = extract_pom_value(pom_document, "parent/artifactId")&.strip
+        parent_version = extract_pom_value(pom_document, "parent/version")&.strip
+        if parent_group_id && parent_artifact_id && parent_version
+          # Recursively call this method for the parent POM document. Parents
+          # are added to the end of the array
+          result += get_pom_documents(parent_group_id, parent_artifact_id, parent_version, depth + 1)
         end
       end
 
-      # merge with parent data if available and take child values on overlap
-      child = {
-        description: extract_pom_value(xml, "description", parent[:properties]),
-        homepage: extract_pom_value(xml, "url", parent[:properties])&.strip,
-        repository_url: repo_fallback(
-          extract_pom_value(xml, "scm/url", parent[:properties])&.strip,
-          extract_pom_value(xml, "url", parent[:properties])&.strip
-        ),
-        licenses: licenses(version_xml).join(","),
-        properties: parent[:properties].merge(extract_pom_properties(xml)),
-      }.select { |_k, v| v.present? }
-
-      parent.merge(child)
+      result
+    end
+
+    def self.mapping_from_pom_documents(pom_documents)
+      properties = {}
+
+      pom_documents.each do |pom_document|
+        properties.merge!(extract_pom_properties(pom_document))
```

Comment:
is this still in the same direction as before (child overwrites parent)?

Reply:
Yep!

Reply:
if it's child->parent then the above would merge parent after child? I think?

Reply:
Fixed in https://github.com/librariesio/libraries.io/pull/3538